cpp/ql/src/semmle/code/cpp/dataflow/internal/DataFlowUtil.qll

```diff
@@ -585,6 +585,23 @@ predicate simpleLocalFlowStep(Node nodeFrom, Node nodeTo) {
     nodeFrom.(PostUpdateNode).getPreUpdateNode().asExpr() = call and
     nodeTo.asDefiningArgument() = call.getQualifier()
   )
+  or
+  // Iterators that have a user-defined `operator=`. Take flow from the RHS to
+  // the post-update node for the iterator.
+  // The built-in `=` case is handled by FlowVar, other user-defined `operator=`
+  // will be handled by interprocedural flow.
+  exists(IteratorPartialDefinitionNode postUpdate, Call opEquals |
+    postUpdate = nodeTo and
+    opEquals.getTarget().hasName("operator=") and
```

Comment:
Without having looked at this in a lot of detail yet - why do we need mentions of `"operator="`, `"operator*"` and `"operator++"` inside `dataflow/internal`?  Shouldn't this kind of thing be done by models?